x-pack/plugins/security_solution/public/common/containers/alerts/use_alert_prevalence.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect, useState } from 'react';
+
+import { DEFAULT_MAX_TABLE_QUERY_SIZE } from '../../../../common/constants';
+import { useGlobalTime } from '../use_global_time';
+import { GenericBuckets } from '../../../../common/search_strategy';
+import { useQueryAlerts } from '../../../detections/containers/detection_engine/alerts/use_query';
+
+const ALERT_PREVALENCE_AGG = 'countOfAlertsWithSameFieldAndValue';
+export const DETECTIONS_ALERTS_COUNT_ID = 'detections-alerts-count';
+
+interface UseAlertPrevalenceOptions {
+  field: string;
+  value: string | string[] | undefined | null;
+  signalIndexName: string | null;
+}
+
+interface UserAlertPrevalenceResult {
+  loading: boolean;
+  count: undefined | number;
+  error: boolean;
+}
+
+export const useAlertPrevalence = ({
+  field,
+  value,
+  signalIndexName,
+}: UseAlertPrevalenceOptions): UserAlertPrevalenceResult => {
+  const { to, from } = useGlobalTime();
+  const [initialQuery] = useState(() => generateAlertPrevalenceQuery(field, value, from, to));
```

Comment:
Should this query also apply the global filters (in the context of the alerts page, for example), or in the context of Timeline, the current timeline's search criteria?

Consider the following screenshot, where the alerts page is filtered by `NOT process.name: notepad.exe`:

![alerts_filtered_by_process_name](https://user-images.githubusercontent.com/4459398/158440831-34e3cc5f-dca2-418d-8390-8f4c450d4cc9.png)

If the user selects the `Show top host.name` hover action shown in the screenshot below:

![show_top_host_name_action](https://user-images.githubusercontent.com/4459398/158441153-eae8e494-ba29-43b9-97bc-95cb31cc99c6.png)

to show the top `host.name` aggregation shown in the screenshot below:

![top_host_name](https://user-images.githubusercontent.com/4459398/158442220-4b062283-0e04-4946-8568-5ad43a002440.png)

the query for the aggregation in the screenshot above includes the filter criteria applied to the page, per the screenshot below:

![inspect_query](https://user-images.githubusercontent.com/4459398/158441762-24bde851-068d-48d9-92c9-5d3e91517443.png)

Should the alert prevalence query include the global / timeline filter criteria in it's query, for consistency with the other aggregations on the page?

Reply:
We discussed offline, but for now we are not applying the filters as it feels like those values shouldn't be "filtered out" For now, we are providing an info tooltip that alludes to this.